client/src/components/PostDashboard.tsx

```diff
@@ -0,0 +1,113 @@
+import Button from "@mui/material/Button";
+import Card from "@mui/material/Card";
+import CardActions from "@mui/material/CardActions";
+import CardContent from "@mui/material/CardContent";
+import CardMedia from "@mui/material/CardMedia";
+import CssBaseline from "@mui/material/CssBaseline";
+import Grid from "@mui/material/Grid";
+import Box from "@mui/material/Box";
+import Container from "@mui/material/Container";
+import { createTheme, ThemeProvider } from "@mui/material/styles";
+import Typography from "@mui/material/Typography";
+import Pagination from "@mui/material/Pagination";
+import Header from "./Header";
+import ViewPostDialog, { GenerateTags } from "./ViewPostDialog";
+import Stack from "@mui/material/Stack";
+
+const cards = [1, 2, 3, 4, 5, 6, 7]; // TODO: We can make this a list of post objects later
+
+const theme = createTheme();
+
+export default function PostDashboard() {
+  return (
+    <ThemeProvider theme={theme}>
+      <CssBaseline />
+      <Header />
+      <main>
+        <Container sx={{ py: 5 }} maxWidth="xl">
+          <Grid container spacing={7}>
+            <Grid
+              item
+              xs={12}
+              style={{
+                display: "flex",
+                justifyContent: "flex-end",
+                alignItems: "flex-end",
+              }}
+            >
+              <Button variant="contained">New Post</Button>
+            </Grid>
+            {cards.map((card) => (
```

Comment:
I would make the card a separate component (`<PostPreview>`?), have it accept either:
1) the post object from the API
2) the post ID and have it pull the data off of [app wide state](https://redux.js.org/)

Reply:
Sure. I'll make the separate component. 

For passing the post object from the API, I think this should be done in a separate PR after the CRUD functions are merged. This will make it easier to pull data from our API and test our code. In this current PR, it's just for the frontend design, so I will leave placeholder text.